runtime/binding-openapi/src/main/java/io/aklivity/zilla/runtime/binding/openapi/internal/streams/OpenapiClientFactory.java

```diff
@@ -407,26 +370,18 @@ private void onOpenapiWindow(
             final long traceId = window.traceId();
             final long budgetId = window.budgetId();
             final int padding = window.padding();
-            final int capabilities = window.capabilities();
 
             assert acknowledge <= sequence;
-            assert sequence <= replySeq;
-            assert acknowledge >= replyAck;
-            assert maximum >= replyMax;
-
-            replyAck = acknowledge;
-            replyMax = maximum;
-            replyBud = budgetId;
-            replyPad = padding;
-            replyCap = capabilities;
-            state = OpenapiState.closingReply(state);
 
-            assert replyAck <= replySeq;
+            state = OpenapiState.closingReply(state);
 
-            http.doHttpWindow(traceId, acknowledge, budgetId, padding);
+            http.doHttpWindow(sequence, acknowledge, maximum, traceId, acknowledge, budgetId, padding);
         }
 
         private void doOpenapiBegin(
+            long replySeq,
+            long replyAck,
+            int replyMax,
```

Comment:
These parameters should not be _directional_, i.e. `sequence`, not `replySeq`, etc.